collector/receiver/otelarrowreceiver/internal/arrow/arrow.go

```diff
@@ -378,6 +379,14 @@ func (r *Receiver) processAndConsume(ctx context.Context, method string, arrowCo
 	ctx, span := r.tracer.Start(ctx, "otel_arrow_stream_recv")
 	defer span.End()
 
+	defer func() {
+		// Set span status if an error is returned.
+		if err != nil {
+			span := trace.SpanFromContext(ctx)
+			span.SetStatus(otelcodes.Error, err.Error())
```

Comment:
Note: this use is correct because the code below is very specific about using the `err` defined above. However, it'd be more defensive for the code to also declare a `retErr` parameter in this case and use it in the deferred funciton, please.